sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/MessageBatch.java

```diff
@@ -0,0 +1,254 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.messaging.servicebus;
+
+import com.azure.core.amqp.AmqpMessageConstant;
+import com.azure.core.amqp.exception.AmqpErrorCondition;
+import com.azure.core.amqp.exception.AmqpException;
+
+import com.azure.core.amqp.implementation.ErrorContextProvider;
+import com.azure.core.amqp.implementation.TracerProvider;
+import com.azure.core.util.Context;
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.core.util.tracing.ProcessKind;
+import org.apache.qpid.proton.Proton;
+import org.apache.qpid.proton.amqp.Binary;
+import org.apache.qpid.proton.amqp.Symbol;
+import org.apache.qpid.proton.amqp.messaging.ApplicationProperties;
+import org.apache.qpid.proton.amqp.messaging.Data;
+import org.apache.qpid.proton.amqp.messaging.MessageAnnotations;
+import reactor.core.publisher.Signal;
+
+import java.nio.BufferOverflowException;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Locale;
+import java.util.Objects;
+import java.util.Optional;
+
+import static com.azure.core.util.tracing.Tracer.DIAGNOSTIC_ID_KEY;
+import static com.azure.core.util.tracing.Tracer.SPAN_CONTEXT_KEY;
+
+/**
+ * A class for aggregating {@link Message} into a single, size-limited, batch. It is treated as a single message when
+ * sent to the Azure Service Bus service.
+ *
+
+ *     producer.
+ */
+ final class MessageBatch {
+    private final ClientLogger logger = new ClientLogger(MessageBatch.class);
+    private final Object lock = new Object();
+    private final int maxMessageSize;
```

Comment:
This is a feature which is requested by customer where SDK calculate max size allowed for a batch message and warn client if their messages are larger than max allowed size.

Reply:
Do we have a concept of message batch in service bus? If this isn't part of a feature we have committed to, we shouldn't add it in.

Reply:
Yes https://docs.microsoft.com/en-us/java/api/com.microsoft.azure.servicebus.queueclient.sendbatch?view=azure-java-stable